packages/cli/src/services/config-service.ts

```diff
@@ -26,6 +26,12 @@ export async function compileProjectAndLoadConfig(userProjectPath: string): Prom
   return readProjectConfig(userProjectPath)
 }
 
+export async function checkAndCompileProject(userProjectPath: string): Promise<BoosterConfig> {
+  await checkItIsABoosterProject(userProjectPath)
+  await compileProject(userProjectPath)
+  return new BoosterConfig('build')
```

Comment:
I think you were forced to return this config object here because of the current interface, but it's weird that you need to do this. I think that you can take advantage from the abstract implementation of the `Script` class to do something like this:

```typescript
Script.init(`boost ${Brand.dangerize('build')} 🚀`, Promise.resolve(process.cwd()))
  .step('Checking project structure', checkItIsABoosterProject)
  .step('Building project', compileProject)
  .info('Build complete!')
  .done()
```

In this case, the type `TContext` will be resolved as `string` and the Script object will be resolved as `Script<string>`, making the action types in the `checkItIsABoosterProject` and `compileProject` match (`(ctx: string) => Promise<void>`. Doing this (and if it works, because I've not tried) you won't need this extra `checkAndCompileProject` function and the final solution will look cleaner. If you try, let me know if you liked this alternate solution and if it worked!

Reply:
OK @javiertoledo . The way you suggest is more convenient